tests/unit/utils/test_log.py

```diff
@@ -204,6 +216,23 @@ def test_stderr_none(self, args):
         log.init_log(args)
         sys.stderr = old_stderr
 
+    @pytest.mark.parametrize('logfilter, negated', [
+        ('!one,two', True),
+        ('one,two', False),
+        ('one,!two', False),
+        (None, False),
+    ])
+    def test_negation_parser(self, args, mocker, logfilter, negated):
+        """Test parsing the --logfilter argument."""
+        filter_mock = mocker.patch('qutebrowser.utils.log.LogFilter',
+                                   autospec=True)
+        args.logfilter = logfilter
+        log.init_log(args)
+        expected_names = (logfilter.lstrip('!').split(',') if logfilter else
+                          None)
```

Comment:
I'd rather list those in `parametrize` instead of reimplementing this in code here - it's less error-prone and serves as a nice documentation. I'll change it while merging though.